packages/neptune-extension/src/common/components/UploadForm.tsx

```diff
@@ -0,0 +1,164 @@
+import { NotebookDTO } from 'generated/leaderboard-client/src/models';
+
+import {
+  PlatformNotebook,
+  PlatformNotebookMetadata,
+} from 'types/platform';
+
+import Modal from 'common/components/Modal';
+
+interface UploadFormProps {
+  platformNotebook: PlatformNotebook
+  notebook: NotebookDTO | undefined
+  defaultProjectId: string
+  onCancel: () => void
+  onSuccess: () => void
+}
+
+const UploadForm: React.FC<UploadFormProps> = ({
+  platformNotebook,
+  notebook,
+  defaultProjectId,
+  onCancel,
+  onSuccess,
+}) => {
+  const [ metadata ] = React.useState(() => platformNotebook.getMetadata());
+
+  const canUploadCheckpoint = notebook
+    && notebook.id === metadata.notebookId;
+    // TODO: check the owner of notebook
+
+  const [ loading, setLoading ] = React.useState(false);
+
+  const [ mode, setMode ] = React.useState(canUploadCheckpoint ? 'checkpoint' : 'notebook');
+
+  const [ projectId, setProjectId ] = React.useState(() => {
+    return !!notebook
+      ? notebook.projectId
+      : defaultProjectId;
+  });
+  const [ name, setName ] = React.useState('');
+  const [ description, setDescription ] = React.useState('');
+
+  function changeMode(newMode: string) {
+    if (newMode === 'checkpoint') {
+      setProjectId((notebook as NotebookDTO).projectId);
+    }
+
+    setMode(newMode);
+  }
+
+  async function uploadCheckpoint(notebookId: string) {
+    const content = await platformNotebook.getContent();
+
+    const checkpoint = await leaderboardClient.api.createEmptyCheckpoint({
+      notebookId,
+      checkpoint: {
+        path: metadata.path,
+        name,
+        description,
+      },
+    });
+
+    await leaderboardClient.api.uploadCheckpointContent({
+      id: checkpoint.id,
+      content,
+    });
+  }
+
+  async function uploadNotebook() {
+    const notebook = await leaderboardClient.api.createNotebook({ projectIdentifier: projectId });
+
+    await uploadCheckpoint(notebook.id);
+
+    await platformNotebook.saveNotebookId(notebook.id);
+  }
+
+  async function handleSubmit() {
+    setLoading(true);
+
+    if (mode === 'notebook') {
+      await uploadNotebook();
+    }
+    else {
+      await uploadCheckpoint(metadata.notebookId as string);
+    }
+
+    setLoading(false);
+
+    onSuccess();
+  }
+
+  const disabled = !projectId;
+
+  return (
+    <Modal
+      isOpen
+      onClose={onCancel}
+    >
+      { (!notebook && metadata.notebookId) && (
+        <span>
+          You have no access to this notebook.
+          A copy of this notebook will be created with current checkpoint as the first one.
+        </span>
+      )}
+
+      { (notebook && notebook.path !== metadata.path) && (
+        <span>
+          The path of this notebook changed.
+          A copy of this notebook will be created with current checkpoint as the first one.
+        </span>
+      )}
+
+      { canUploadCheckpoint && (
+        <React.Fragment>
+          <a href="#" onClick={() => changeMode('notebook')}>
```

Comment:
It was supposed to be a radio button, not a link.
It would be better to use radio as it will have most likely the same interface as final UI component.